.github/workflows/release-prep.yaml

```diff
@@ -0,0 +1,71 @@
+name: Prep Heat Release
+
+on:
+  issues:
+    types: [opened]
+env:
+  working_branch: workflows/release-prep
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@91182cccc01eb5e619899d80e4e971d6181294a7 # v2.10.1
+        with:
+          egress-policy: audit
+
+      - uses: actions/checkout@v3
+      - name: Extract Version Information
+        id: extract-version
+        if: contains(github.event.issue.labels.*.name, 'release-prep')
+        run: |
+          issue_title="${{ github.event.issue.title }}"
+          major=$(echo "$issue_title" | cut -d '.' -f 1)
+          minor=$(echo "$issue_title" | cut -d '.' -f 2)
+          micro=$(echo "$issue_title" | cut -d '.' -f 3)
+          echo "major=$major" >> $GITHUB_OUTPUT
+          echo "minor=$minor" >> $GITHUB_OUTPUT
+          echo "micro=$micro" >> $GITHUB_OUTPUT
+      - name: Delete Existing Version Update Branch
+        run: git push --delete origin ${{ env.working_branch }} || true
+      - name: Create Release Branch
+        if: steps.extract-version.outputs.micro == 0
+        run: |
+          git checkout -b release/${{ steps.extract-version.outputs.major }}.${{ steps.extract-version.outputs.minor }}.x
+          git push origin release/${{ steps.extract-version.outputs.major }}.${{ steps.extract-version.outputs.minor }}.x
+      - name: Bump Version
+        if: always()
+        run: |
+          git fetch origin
+          git checkout release/${{ steps.extract-version.outputs.major }}.${{ steps.extract-version.outputs.minor }}.x
+          git checkout -b ${{ env.working_branch }}
+          git push origin ${{ env.working_branch }}
+          sed -i "s/major: int = \([0-9]\+\)/major: int = ${{ steps.extract-version.outputs.major }}/g" heat/core/version.py
+          sed -i "s/minor: int = \([0-9]\+\)/minor: int = ${{ steps.extract-version.outputs.minor }}/g" heat/core/version.py
+          sed -i "s/micro: int = \([0-9]\+\)/micro: int = ${{ steps.extract-version.outputs.micro }}/g" heat/core/version.py
+          sed -i "s/extension: str = .*/extension: str = None/g" heat/core/version.py
+      - name: Create PR from branch
+        uses: peter-evans/create-pull-request@5e914681df9dc83aa4e4905692ca88beb2f9e91f # v7.0.5
+        with:
+            base: release/${{ steps.extract-version.outputs.major }}.${{ steps.extract-version.outputs.minor }}.x
+            branch: ${{ env.working_branch }}
```

Comment:
The base branch should be "release", and the branch should be our newly created branch, right? "pre-release/x.y.z"